test/functorch/test_vmap.py

```diff
@@ -1435,6 +1435,15 @@ def clone_contiguous(x):
         with self.assertRaisesRegex(RuntimeError, msg):
             vmap(lambda x: x.clone(memory_format=torch.channels_last_3d))(torch.randn(B0))
 
+    def test_decomposition_under_python_dispatcher(self):
```

Comment:
I have checked locally that the test passes locally even without the fix.

I think TestVmapBase doesn't do what the below comment says. Probably because the `torch._C._functorch._is_vmap_fallback_enabled()` is True by default.

https://github.com/pytorch/pytorch/blob/a52cf3398c90d0407f77cce392173d597c981b7e/test/functorch/test_vmap.py#L1269-L1272

**NOTE:** Adding `torch._C._functorch._set_vmap_fallback_enabled(False)` right after the imports in this file leads to a bunch of failure. Will file an issue for the same.

Reply:
Thanks for catching that. I've added a manual test to check for the vmap fallback warning and verified it fails before this patch. We probably disabled the automatic assertion on the error at some point in the past and never turned it back on...